### ISSUE TITLE
Auto-importing for some items doesn't work

### ISSUE THREAD

**Issue by natefaubion** [nwolverson/purescript-language-server#99]
In the latest release I'm having trouble with many things not auto-importing.

* It appears type synonyms are always importing with DCtor filter, which they aren't and so always fail.
* It appears terms like `_DbQuery` (which is a value declaration of a prism) parse as a `Type` filter, so it always fails.

The log is very noisy and appears to be logging the entire file contents, instead of just what it's trying to import.

**Comment by natefaubion** [comment]
Downgrading to the previous release fixes these issues.

**Comment by andys8** [comment]
Same here: https://github.com/nwolverson/purescript-language-server/issues/97

**Comment by nwolverson** [comment]
I suspected this was doomed to failure. I did test with various things like type synonyms but obviously not great coverage.

Previously we were importing all unfiltered, leading to random results between a type and its data constructor, but the relevant information I think we have is the type/kind of the identifier (and of course its name).

I suspect the identifier namespace should just be added to the IDE server response but I wonder if someone more intelligent can come up with a proper heuristic before then - shame to wait for a compiler release.

(The logging is unfortunate, pardon me - I typod the document text for the import identifier in the log... - going to push a quick release that turns off this filtering and stomps that log)
